mvlearn/ajive/ajive.py

```diff
@@ -0,0 +1,760 @@
+                 joint_rank=None, indiv_ranks=None,
+                 center=True,
+                 reconsider_joint_components=True,
+                 wedin_percentile=5, n_wedin_samples=1000,
+                 precomp_wedin_samples=None,
+                 randdir_percentile=95, n_randdir_samples=1000,
+                 precomp_randdir_samples=None,
+                 store_full=True, n_jobs=None):
+
+        self.init_signal_ranks = init_signal_ranks
+        self.joint_rank = joint_rank
+        self.indiv_ranks = indiv_ranks
+
+        self.center = center
+
+        self.wedin_percentile = wedin_percentile
+        self.n_wedin_samples = n_wedin_samples
+        self.wedin_samples_ = precomp_wedin_samples
+        if precomp_wedin_samples is not None:
+            self.n_wedin_samples = len(list(precomp_wedin_samples.values())[0])
+
+        self.randdir_percentile = randdir_percentile
+        self.n_randdir_samples = n_randdir_samples
+        self.random_sv_samples_ = precomp_randdir_samples
+        if precomp_randdir_samples is not None:
+            self.n_randdir_samples = len(precomp_randdir_samples)
+
+        self.reconsider_joint_components = reconsider_joint_components
+
+        self.store_full = store_full
+
+        self.n_jobs = n_jobs
+
+    def __repr__(self):
+
+        if self.is_fit:
+            r = "joint rank: {}".format(self.common.rank)
+            for bn in self.block_names:
+                indiv_rank = self.blocks[bn].individual.rank
+                r += ", block {} indiv rank: {}".format(bn, indiv_rank)
+            return r
+
+        else:
+            return "No data has been fitted yet"
+
+    def fit(self, blocks, precomp_init_svd=None):
+        """
+        Fits the AJIVE decomposition.
+
+        Parameters
+        ----------
+        blocks: {list of array-likes or numpy.ndarray, dict}
+            - blocks length: n_views
+            - blocks[i] shape: (n_samples, n_features_i) 
+            The different views that are input. Input as data matrices. 
+            If dict, will name blocks by keys, otherwise blocks are named by 
+            0, 1, ...K. 
+            
+        precomp_init_svd: {list, dict, None}, optional
+            Precomputed initial SVD. Must have one entry for each data block.
+            The SVD should be a 3 tuple (scores, svals, loadings), see output
+            of .svd_wrapper for formatting details.
+
+        """
+        blocks, self.init_signal_ranks, self.indiv_ranks, precomp_init_svd,\
+            self.center, obs_names, var_names, self.shapes_ = \
+                _arg_checker(blocks,
+                            self.init_signal_ranks,
+                            self.joint_rank,
+                            self.indiv_ranks,
+                            precomp_init_svd,
+                            self.center)
+
+        block_names = list(blocks.keys())
+        num_obs = list(blocks.values())[0].shape[0] #number of views
+
```

Comment:
you import check_Xs, but you don't use it. Is there any way to validate the input you are getting? For example, if you need there to be exactly 2 views, check_Xs can be used to do that if you first convert the blocks to a temporary list of matrices